Web Application Codes/web/src/components/MapsPage.js

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useState } from 'react';
+import GoogleMapReact from 'google-map-react';
+import AutoComplete from '../AutoComplete';
+import Marker from './Marker';
+import './Sidenav/sidenav.css';
+import Pin from '../icons/location-pin.svg';
+// import { Icon } from '@iconify/react'
+// import locationIcon from '@iconify/icons-mdi/map-marker'
+// import {
+//   GoogleMapProvider,
+//   HeatMap,
+//   InWindow,
+//   MapBox,
+//   Marker,
+//   OverlayView,
+//   Polygon,
+// } from '@googlemap-react/core'
+
+const MapsPage = () => {
+	const [lat, setLat] = useState(9.082);
+	const [lng, setLng] = useState(8.6753);
+	const [zoom, setZoom] = useState(2);
+	// const [google, setGoogle] = useState();
+	const [mapApiLoaded, setMapApiLoaded] = useState(false);
+	const [mapApi, setMapApi] = useState();
+	const [mapInstance, setMapInstance] = useState();
+
+	const [places, setPlaces] = useState([]);
+
+	useEffect(() => {
+		if (places.length === 3) {
+			setPlaces([]);
+		}
+		console.log(Pin);
+	}, [places]);
+
+	const handleClick = ({ lat, lng }) => {
+		console.log(lat, lng);
+		setLat(lat);
+		setLng(lng);
+		setZoom(zoom + 2);
+	};
+	console.log(places.length);
+	const handleApiLoaded = (map, maps) => {
+		setMapApi(maps);
+		setMapInstance(map);
+		setMapApiLoaded(true);
+	};
+	// const [stateInfo, setStateInfo] = useState([]);
+
```

Comment:
commented code :fearful:

Reply:
while you may remove the commented code, please make it more readable by adding explanatory comments